### PR TITLE
Binary upload for react native and new activity SalesforceReactActivity

### PR DESCRIPTION
**Pull request by wmathurin**

### CODE REVIEW

File: libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/ui/SalesforceReactActivity.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import com.salesforce.androidsdk.rest.RestClient;
+import com.salesforce.androidsdk.security.PasscodeManager;
+import com.salesforce.androidsdk.util.EventsObservable;
+import com.salesforce.androidsdk.util.EventsObservable.EventType;
+import com.salesforce.androidsdk.util.UserSwitchReceiver;
+
+/**
+ * Super class for all Salesforce activities.
+ */
+public abstract class SalesforceReactActivity extends ReactActivity {
```
Comment (wmathurin) [inline libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/ui/SalesforceReactActivity.java]: Extends ReactActivity but behaves like a SalesforceActivity, react native template app's MainActivity extends SalesforceReactActivity
Previously react native app's MainActivity was extending SalesforceActivity and was kind of behaving like a ReactActivity but not very well.

**Comment by bhariharan** [comment]
:+1:

**Comment by kchitalia** [comment]
Do we have to update the sample app?

**Comment by wmathurin** [comment]
Yep working on it.

>

**Pull request merged**